### PR TITLE
libdigidocpp: Fix PKCS11 module library path

### PR DESCRIPTION
**Pull request by klemensn**
By default, the OpenSC module is loaded as relative filename, i.e.
"opensc-pkcs11.so" is searched for in the default library path.

Point `LD_LIBRARY_PATH` for `digidoc-tool` to the OpenSC package just
like pkgs/tools/seurity/qdigidoc/deafult.nix already does.

This makes `digidoc-tool` work by default without manually setting
`LD_LIBRARY_PATH` or passing `--pkcs11=/path/to/module`.

@jagajaga @flokli

### CODE REVIEW

File: pkgs/development/libraries/libdigidocpp/default.nix
Comment (flokli) [inline pkgs/development/libraries/libdigidocpp/default.nix:22]: Prefixing `LD_LIBRARY_PATH` is messy, as it propagates processes. Check if you can set `PKCS11_ENGINE=` to an absolute path (like in `pkgs/tools/security/rng-tools/default.nix`), or if it's possible to update the path that's loaded at runtime to an absolute one (like in `pkgs/tools/security/chrome-token-signing/default.nix`).

If one of these is possible, the same approach should be applied to `pkgs/tools/security/qdigidoc/default.nix`, too.

**Review by flokli — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/libdigidocpp/default.nix
Comment (flokli) [reply pkgs/development/libraries/libdigidocpp/default.nix:22]: In any case, this should at least include a comment with a link to the upstream PR, so people can track the upstreaming status.

In the case of this application here, all that's needed should be a

```
+  cmakeFlags = [
+    "-DPKCS11_MODULE=${opensc}/lib/opensc-pkcs11.so"
+  ];
```

This currently doesn't work on Linux, as the upstream `FindPKCS11.cmake` unconditionally sets the value - I sent a PR in https://github.com/open-eid/cmake/pull/35 to fix this, which seems a bit cleaner than just setting a random environment variable during the build.

Comment (klemensn) [reply pkgs/development/libraries/libdigidocpp/default.nix:22]: I'd say `PKCS11_MODULE_PATH` is as random as `PKCS11_MODULE`...

Both PRs work as expected and I don't mind either way, so I'll leave that to upstream.

I'll add a link so nixpkgs hackers will see it, although I dare say one ought to look through upstream changes anyway
when updating a package (just like basic functionality should be tested....).

**Comment by klemensn** [comment]
Rebased and conflicts resolved after #175376 was merged.

**Pull request merged**
